### PR TITLE
radicle-surf: add CHANGELOG.md

### PR DESCRIPTION
**Pull request by keepsimple1**
This is for one of tasks in #100 .

There are too many changes in our rewrite to list them all. I tried to provide a summary with some highlights and then refer to our cargo doc. Hopefully that makes sense.

### CODE REVIEW

File: radicle-surf/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+changes include:
+
+- `Browser` is removed. Its methods are implemented directly with `Repository`.
+- `vcs` is removed. No more generic VCS. Only `git` is supported.
+- `Ref` and `RefScope` are removed. Re-use `git-ref-format` crate and a new
+`glob` for the ref patterns.
+- Added support of `Tree` and `Blob` that correspond to their defintions in git.
+- Added two new traits `Revision` and `ToCommit` that make methods flexiable and
+still simple to use.
+- The module hierarchy is much more flat than before.
```
Comment (FintanH) [inline radicle-surf/CHANGELOG.md:14]: I don't think we need to mention this tbh
```suggestion
```

File: radicle-surf/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+## Version 0.9.0
+
+This release consists of a major rewrite of this crate. Its API is overall
+simplified and is not compatible with the previous version (v0.8.0). The main
+changes include:
+
+- `Browser` is removed. Its methods are implemented directly with `Repository`.
+- `vcs` is removed. No more generic VCS. Only `git` is supported.
```
Comment (FintanH) [inline radicle-surf/CHANGELOG.md:8]:
```suggestion
- Git will be the only supported VCS. Any extension points for other VCSes were removed.
```

File: radicle-surf/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- `Ref` and `RefScope` are removed. Re-use `git-ref-format` crate and a new
```
Comment (FintanH) [inline radicle-surf/CHANGELOG.md:9]:
```suggestion
- `Ref` and `RefScope` are removed. Re-use the `git-ref-format` crate and a new
```

File: radicle-surf/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+`glob` for the ref patterns.
+- Added support of `Tree` and `Blob` that correspond to their defintions in git.
```
Comment (FintanH) [inline radicle-surf/CHANGELOG.md:11]:
```suggestion
- Added support of `Tree` and `Blob` that correspond to their definitions in Git.
```

**Review by FintanH — COMMENTED** [review]
Can we format it to 80 column width, please? :)

I think the points are good, but I suggested some changes :+1:

File: radicle-surf/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+## Version 0.9.0
```
[the remark this answers is not in the document]
Comment (keepsimple1) [reply radicle-surf/CHANGELOG.md:1]: I'm okay with either v0.9 or v1.0. I bumped it only to  0.9.0 as I didn't feel the API is mature enough for 1.0.0, yet.

I will update `Cargo.toml`.

**Comment by keepsimple1** [comment]
Thanks for the corrections. Sorry about the spelling errors. I need to find a spelling plugin ;-). 

> Can we format it to 80 column width, please? :)

I did so in VSCode, maybe something weird is going on.  Will check again.

**Pull request merged**